datadog/resource_datadog_dashboard.go

```diff
@@ -7613,6 +7795,343 @@ func buildDatadogTraceServiceDefinition(terraformDefinition map[string]interface
+	datadogStaticSplits := make([][]datadogV1.SplitVectorEntryItem, len(terraformStaticSplits))
+	//going over each static split
+	for i, terraformStaticSplit := range terraformStaticSplits {
+		terraformStaticSplitMap := terraformStaticSplit.(map[string]interface{})
+		//building inner array for static split from terraform split vector list.
+		for _, splitVector := range terraformStaticSplitMap["split_vector"].([]interface{}) {
+			datadogSplitVectorMap := splitVector.(map[string]interface{})
+
+			datadogSplitVector := datadogV1.SplitVectorEntryItem{}
+			if v, ok := datadogSplitVectorMap["tag_key"].(string); ok {
+				datadogSplitVector.SetTagKey(v)
+			}
+			if tagValuesList, ok := datadogSplitVectorMap["tag_values"].([]interface{}); ok {
+				datadogTagValues := make([]string, len(tagValuesList))
+				for k, tagValues := range tagValuesList {
+					datadogTagValues[k] = tagValues.(string)
+				}
+				datadogSplitVector.SetTagValues(datadogTagValues)
+			}
+			datadogStaticSplits[i] = append(datadogStaticSplits[i], datadogSplitVector)
+		}
+	}
+	return &datadogStaticSplits
+}
+
+func buildTerraformSplitGraphDefinition(datadogDefinition *datadogV1.SplitGraphWidgetDefinition) map[string]interface{} {
+	terraformDefinition := map[string]interface{}{}
+	// Required params
+	if v, ok := datadogDefinition.GetSourceWidgetDefinitionOk(); ok {
+		terraformDefinition["source_widget_definition"] = []map[string]interface{}{buildTerraformSourceWidgetDefinition(v)}
+	}
+	if v, ok := datadogDefinition.GetSizeOk(); ok {
+		terraformDefinition["size"] = v
+	}
+	if v, ok := datadogDefinition.GetSplitConfigOk(); ok {
+		terraformDefinition["split_config"] = []map[string]interface{}{*buildTerraformSplitConfig(v)}
+	}
+	// Optional params
+	if v, ok := datadogDefinition.GetHasUniformYAxesOk(); ok {
+		terraformDefinition["has_uniform_y_axes"] = *v
+	}
+	if v, ok := datadogDefinition.GetTitleOk(); ok {
+		terraformDefinition["title"] = *v
+	}
+	if v, ok := datadogDefinition.GetTimeOk(); ok {
+		terraformDefinition["live_span"] = v.GetLiveSpan()
+	}
+
+	return terraformDefinition
+}
+
+func buildTerraformSplitConfig(datadogSplitConfig *datadogV1.SplitConfig) *map[string]interface{} {
+	terraformSplitConfig := map[string]interface{}{}
+
+	if v, ok := datadogSplitConfig.GetSplitDimensionsOk(); ok {
+		datadogSplitDimensions := *v
+		terraformOneGraphPer := map[string]interface{}{}
+
+		terraformOneGraphPer["one_graph_per"] = datadogSplitDimensions[0].OneGraphPer
+		terraformSplitConfig["split_dimensions"] = []map[string]interface{}{terraformOneGraphPer}
+	}
+
+	if v, ok := datadogSplitConfig.GetLimitOk(); ok {
+		terraformSplitConfig["limit"] = *v
+	}
+
+	if datadogSort, ok := datadogSplitConfig.GetSortOk(); ok {
+		terraformSortList := []map[string]interface{}{
+			{
+				"order": datadogSort.Order,
+				"compute": []map[string]interface{}{
+					{
+						"aggregation": datadogSort.Compute.Aggregation,
+						"metric":      datadogSort.Compute.Metric,
```

Comment:
these lines will segfault with nil pointer dereference if `Compute` is missing from the server response (which I confirmed was possible with a test dashboard, and allowed since it is not required in the spec)